File: common/src/main/java/dev/chasem/cobblemonextras/commands/PokeSee.java
package dev.chasem.cobblemonextras.commands;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.context.CommandContext;
import dev.chasem.cobblemonextras.CobblemonExtras;
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions;
import dev.chasem.cobblemonextras.screen.PokeSeeHandlerFactory;
import net.minecraft.command.argument.EntityArgumentType;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;

import static net.minecraft.server.command.CommandManager.argument;
import static net.minecraft.server.command.CommandManager.literal;

public class PokeSee {

    public void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(
                literal("pokesee")
                        .requires(src -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKESEE_PERMISSION))
                        .executes(this::self)
        );
        dispatcher.register(
                literal("pokeseeother")
                        .requires(src -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKESEE_OTHER_PERMISSION))
                        .then(argument("player", EntityArgumentType.player())
                                .executes(this::other))
        );
    }

    private int self(CommandContext<ServerCommandSource> ctx) {
        if (ctx.getSource().getPlayer() != null) {
            ServerPlayerEntity player = ctx.getSource().getPlayer();
            player.openHandledScreen(new PokeSeeHandlerFactory());
        }
        return 1;
    }
    private int other(CommandContext<ServerCommandSource> ctx) {
        if (ctx.getSource().getPlayer() != null) {
            ServerPlayerEntity player = ctx.getSource().getPlayer();
            String otherPlayerName = ctx.getInput().split(" ")[1]; // ctx.getArgument("player", String.class);
            ServerPlayerEntity otherPlayer = ctx.getSource().getServer().getPlayerManager().getPlayer(otherPlayerName);
            if (otherPlayer != null) {
                player.openHandledScreen(new PokeSeeHandlerFactory(otherPlayer));
            }
        }
        return 1;
    }

}

File: common/src/main/kotlin/dev/chasem/cobblemonextras/commands/PokeSee.kt
package dev.chasem.cobblemonextras.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import dev.chasem.cobblemonextras.CobblemonExtras
import dev.chasem.cobblemonextras.menus.PokeSeeMenu
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.commands.arguments.EntityArgument
import net.minecraft.commands.arguments.selector.EntitySelector
import net.minecraft.server.level.ServerPlayer

class PokeSee {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
                Commands.literal("pokesee")
                        .requires { src: CommandSourceStack? -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKESEE_PERMISSION) }
                        .executes { ctx: CommandContext<CommandSourceStack> -> this.self(ctx) }
        )
        dispatcher.register(
                Commands.literal("pokeseeother")
                        .requires { src: CommandSourceStack? -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKESEE_OTHER_PERMISSION) }
                        .then(Commands.argument<EntitySelector>("player", EntityArgument.player())
                                .executes { ctx: CommandContext<CommandSourceStack> -> this.other(ctx) })
        )
    }

    private fun self(ctx: CommandContext<CommandSourceStack>): Int {
        if (ctx.source.player != null) {
            val player: ServerPlayer = ctx.source.player!!
            player.openMenu(PokeSeeMenu(player))
        }
        return 1
    }

    private fun other(ctx: CommandContext<CommandSourceStack>): Int {
        if (ctx.source.player != null) {
            val player: ServerPlayer = ctx.source.player!!
            val otherPlayerName: String = ctx.input.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray().get(1) // ctx.getArgument("player", String.class);
            val otherPlayer: ServerPlayer? = ctx.source.server.playerList.getPlayerByName(otherPlayerName)
            if (otherPlayer != null) {
                player.openMenu(PokeSeeMenu(otherPlayer))
            }
        }
        return 1
    }
}
